animation_nodes/algorithms/mesh/triangulate_mesh.pyx

```diff
@@ -0,0 +1,353 @@
+        polyLength = oldPolyLengths[i]
+
+        # Initialization of polygon's vertices.
+        verticesData.head = 0
+        verticesData.data = <Vertex*>PyMem_Malloc(polyLength * sizeof(Vertex))
+
+        vertexData = verticesData.data
+        for j in range(1, polyLength - 1):
+            vertexData[j].index = oldIndices[polyStart + j]
+            vertexData[j].location = vertices.data[oldIndices[polyStart + j]]
+            vertexData[j].previous = j - 1
+            vertexData[j].next = j + 1
+
+        index = 0
+        vertexData[index].index = oldIndices[polyStart + index]
+        vertexData[index].location = vertices.data[oldIndices[polyStart + index]]
+        vertexData[index].previous = polyLength - 1
+        vertexData[index].next = 1
+
+        index = polyLength - 1
+        vertexData[index].index = oldIndices[polyStart + index]
+        vertexData[index].location = vertices.data[oldIndices[polyStart + index]]
+        vertexData[index].previous = polyLength - 2
+        vertexData[index].next = 0
+
+        # Project vertices in xy-plane.
+        projectPolygonVertices(verticesData)
+
+        # Check polygon polarity.
+        makeSurePolygonIsCounterClockwise(verticesData)
+
+        # Calculate angle and status of vertices.
+        for j in range(polyLength):
+            vertexData[j].angle = calculateAngleWithSign(vertexData, j)
+            setVertexEarStatus(verticesData, j)
+
+        # Calculate triangle polygon indices.
+        for j in range(polyLength - 3):
+            # Removing an ear which has smallest inner-angle.
+            earIndex = findEarHasMinAngle(verticesData)
+
+            newPolyStarts[triangleIndex] = polyIndex
+            newPolyLengths[triangleIndex] = 3
+            triangleIndex += 1
+
+            newIndices[polyIndex] = vertexData[vertexData[earIndex].previous].index
+            newIndices[polyIndex + 1] = vertexData[earIndex].index
+            newIndices[polyIndex + 2] = vertexData[vertexData[earIndex].next].index
+            polyIndex += 3
+
+            if earIndex == verticesData.head: verticesData.head = vertexData[earIndex].next
+            removeEarVertex(verticesData, earIndex)
+
+        newPolyStarts[triangleIndex] = polyIndex
+        newPolyLengths[triangleIndex] = 3
+        triangleIndex += 1
+
+        index = 0
+        headIndex = verticesData.head
+        currentIndex = headIndex
+        while True:
+            newIndices[polyIndex + index] = vertexData[currentIndex].index
+            currentIndex = vertexData[currentIndex].next
+            index += 1
+            if currentIndex == headIndex: break
+        polyIndex += 3
+
+        PyMem_Free(vertexData)
+
+    return newPolygons
+
+# Find the index of minimum angle.
+cdef int findEarHasMinAngle(Vertices verticesData):
+    cdef Py_ssize_t headIndex = verticesData.head
+    cdef Vertex* vertexData = verticesData.data
+    cdef Py_ssize_t currentIndex
+
+    cdef float angleMin = 3.14
```

Comment:
`angleMin` should be the angle of the head vertex.

Reply:
Okay.

Reply:
Omar, we can't set it as the angle of the head-vertex because head-vertex may not be an ear but can have small angle.

Reply:
I see. Then you should increase the value to 3.15, and add a comment explaining this.

Reply:
Okay.

Reply:
Done.